### PR TITLE
Cross-link hasIndices docs

### LINKED ISSUE #21957: RegExp.hasIndices() - more examples and named capture groups
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/hasIndices

### What specific section or headline is this issue about?

Examples and Description

### What information was incorrect, unhelpful, or incomplete?

From engineer in [this bugzilla post](https://bugzilla.mozilla.org/show_bug.cgi?id=1519483#c14):

> The only thing I would consider adding is a more prominent description (maybe on the hasIndices page?) of the structure of the output, similar to the examples [here](https://github.com/tc39/proposal-regexp-match-indices#examples).

> In particular, it might be nice to document the interaction between /d and named capture groups.

### What did you expect to see?

Any information about interaction between /d and named capture groups.

### Do you have any supporting links, references, or citations?

https://github.com/tc39/proposal-regexp-match-indices#examples

### Do you have anything more you want to share?

Once resolution of this is determined and done (or not done) we should remove `dev-doc-needed` from https://bugzilla.mozilla.org/show_bug.cgi?id=1519483

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/regexp/hasindices`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/hasIndices
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/regexp/hasindices/index.md
* Last commit: https://github.com/mdn/content/commit/8db2031e46de297b9a02dca59eb8d8767fe6c61e
* Document last modified: 2022-10-23T08:06:12.000Z

</details>

### PR DESCRIPTION
**Pull request by Josh-Cena**
### Description

Fix #21957

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/regexp/hasindices/index.md [hunk 1]
```diff
@@ -21,10 +21,14 @@ The **`hasIndices`** accessor property indicates whether or not the `d` flag is
...
 
+This flag primarily affects the return value of [`exec()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec). If the `d` flag is present, the array returned by `exec()` has an additional `indices` property as described in the `exec()` method's [return value](/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec#return_value). Because all other regex-related methods (such as {{jsxref("String.prototype.match()")}}) call `exec()` internally, they will also return the indices if the regex has the `d` flag.
+
 The set accessor of `hasIndices` is `undefined`. You cannot change this property directly.
 
 ## Examples
 
+There a more detailed usage example at [Groups and backreferences > Using groups and match indices](/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Groups_and_Backreferences#using_groups_and_match_indices).
```
Comment (hamishwillee) [inline files/en-us/web/javascript/reference/global_objects/regexp/hasindices/index.md:30]: I moved the comment about another example into the examples section. I'll approve so you can merge if you're OK with this, or revert if you don't like it.

**Review by hamishwillee — APPROVED** [review]
Yes, this is helpful. Thank you.

File: files/en-us/web/javascript/reference/global_objects/regexp/hasindices/index.md (hunk 1 above)
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/regexp/hasindices/index.md:30]:
```suggestion
There's a more detailed usage example at [Groups and backreferences > Using groups and match indices](/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Groups_and_Backreferences#using_groups_and_match_indices).
```

Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/regexp/hasindices/index.md]: Sounds good to me 👍

**Comment by hamishwillee** [comment]
THanks very much!

**Pull request merged**